runtime/src/commonMain/kotlin/pbandk/FieldDescriptor.kt

```diff
@@ -1,16 +1,23 @@
 package pbandk
 
+import kotlin.reflect.KProperty0
 import kotlin.reflect.KProperty1
 
 @PublicForGeneratedCode
-class FieldDescriptor<T>(
+class FieldDescriptor<M : Message, T>(
+    messageDescriptor: KProperty0<MessageDescriptor<M>>,
     val name: String,
     val number: Int,
     val type: Type,
-    val value: KProperty1<*, T>,
+    val value: KProperty1<M, T>,
     val oneofMember: Boolean = false,
     val jsonName: String? = null
 ) {
+    // At the time that the [FieldDescriptor] constructor is called, the parent [MessageDescriptor] has not been
+    // constructed yet. This is because this [FieldDescriptor] is one of the parameters that will be passed to the
+    // [MessageDescriptor] constructor. To avoid the circular dependency, this property is declared lazy.
+    val messageDescriptor: MessageDescriptor<M> by lazy { messageDescriptor.get() }
```

Comment:
Thanks for this comment, the lazy makes sense. I am curious about the purpose of this field. Once the object is fully constructed, what's the use case of using this field? I'm looking at it from the perspective of `FieldDescriptor` but perhaps i'm missing the larger picture.

Reply:
We admittedly don't have a use case for it today. So I could be convinced to leave it off until we need it. But in general it can be useful to access the descriptor of the message that a field is defined in. It's basically reflection but at the protobuf level rather than the Java/Kotlin level. And since this is populated from generated code, it's one of those things that's harder to add later in a backwards-compatible way without requiring the generated code to be regenerated.

For comparison, the official Protobuf Java library has something similar: https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/Descriptors.FieldDescriptor.html#getContainingType--.

Reply:
Ah okay. The `getContainingType` example really helped to understand potential needs for this. Thanks!